### DISCUSSION TITLE
Implement add to Bring! shopping list

### DISCUSSION THREAD

**Discussion by ykorzikowski** [mealie-recipes/mealie#790] [Feature Request]
**Before submitting this feature request I have**
_COMPLETE AND FILL THESE OUT_
- [x] Reviewed the documentation and verified that my feature request isn't already a feature.
- [x] Reviewed the existing feature requests and my feature has not been requested.
- [x] Checked the [tasks taged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

**Please Describe The Problem To Be Solved**
Implement the possibility to add a receipes ingredients to external shopping list Bring! (https://www.getbring.com/)

You can find developer page here: https://www.getbring.com/de/integration-prufen and https://sites.google.com/getbring.com/bring-import-dev-guide/web-to-app-integration

**(Optional): Suggest A Solution**

Receipes need to follow this schema: https://schema.org/ingredients

Then, the bring app should be able to parse the ingredients and the receipe properly. 

**Additional Information**

- [ ] If the feature is accepted I'm willing to submit a PR to provide this feature

**Comment by ykorzikowski** [comment]
Example given: chefkoch.de is using this schema and has the possibility to be parsed by bring

```
<div class="recipe-servings ds-box">
    <form class="ds-inline-form bi-form-servings i-amphtml-form" action="https://www.chefkoch.de/rezepte/1182801224334883/1A-Gnocchi-mit-Scampi.html" target="_top" method="get" novalidate="">
    <input aria-label="Anzahl der Portionen" class="ds-input" type="number" step="any" min="0.01" max="1000" name="portionen" autocomplete="off" value="2">

    <button class="ds-btn ds-btn-primary ds-btn-icon" type="submit">
        <i class="material-icons"></i>
        <span>Portionen<span class="ds-sr-only"> umrechnen</span></span>
    </button>
</form>
</div>
```

```
    <table class="ingredients table-header" width="100%" cellspacing="0" cellpadding="0">
                <tbody>
                                            <tr>
                                        <td class="td-left " width="33%"><span>
                                1                                Pck.                            </span></td>
                    <td class="td-right" width="66%">
                                                    <span>Gnocchi </span>
                                            </td>
                </tr>
```

(see https://www.chefkoch.de/rezepte/1182801224334883/1A-Gnocchi-mit-Scampi.html)

So maybe this feature can be resolved by ading those schema-tags to the class attribute.

(see https://food.korzikowski.de/recipe/1a-gnocchi-mit-scampi)

**Comment by TheHesster** [comment]
+1 request for this feature. I wish I knew more about how to code to help out...

**Comment by Z30-Z30** [comment]
+1 Me and my friends are using Bring! a lot. It's handy because you get also information about deals on the markets in the region.

**Comment by ulimanaio** [comment]
Adding a button like this to each recipe: (Add ingredients to Bring!)
![image](https://github.com/mealie-recipes/mealie/assets/95037770/9cde0707-c8b9-4901-9782-92feadf92e5a)
would be amazing. I personally do not need synchronization of both lists, as I mainly use Bring when in the Supermarket.
In general there are a lot of requests for Bring integration - Thank you very much to the developers for looking at integration! Great work :)

**Comment by ZerhackerJagi** [comment]
+1 for this feature. A few more word on this: In my experience the "add to bring" button doesn't add the ingredient quantity if there's more then one recipe using e.g. tomatoes. It would just add the quantity as text like: "tomatoes 1+4+3". Adding ingredients to a mealie shopping list works great. So my personal preference would be to add the bring button to the mealie shopping list as an export option.

**Comment by ulimanaio** [comment]
Is there a way to sponsor this feature? I am not sure if there is a roadmap and if it is on there but I would be willing to support the feature with a coffee or two :)

**Comment by Hubobel** [comment]
Please bring this feature!!!!

**Comment by blickwerker** [comment]
This is hot! Please consider adding this feature.

**Comment by lz039** [comment]
I thought about syncing the lists via API, but this would be much easier.
According to this test on bring website the recipes are already in the correct format, only the script needs to be added.
![image](https://github.com/mealie-recipes/mealie/assets/101728078/96abed9b-9c65-4955-b88a-249d5fd787ca)

I'm not quite sure, because if I try the alternative with deeplinking
```
<a href="https://api.getbring.com/rest/bringrecipes/deeplink?url={the url of your recipe}&source=web&baseQuantity=4&requestedQuantity=4">
{your rendering of the button comes here}
</a>
```
I get the recipe is not compatible.

**Comment by michael-genson** [comment]
This will be possible with https://github.com/mealie-recipes/mealie/pull/3448

**Comment by TheHesster** [comment]
Exciting!

**Comment by michael-genson** [comment]
https://github.com/mealie-recipes/mealie/pull/3448 has been merged into nightly. For more information on how to integrate with Bring! shopping list, [please check out the Mealie docs](https://docs.mealie.io/documentation/getting-started/features/#recipe-actions).
